### PR TITLE
[WFCORE-7036] TransformationUtils.modelToResource() should not return…

### PR DESCRIPTION
**Pull request by kabir**
… undefined if possible children are not defined

https://issues.redhat.com/browse/WFCORE-7036

https://github.com/wildfly/wildfly/pull/18321 must be merged first for the full integration jobs can pass. 
This is needed for the transformers tests in https://github.com/wildfly/wildfly/pull/18320 to pass

### CODE REVIEW

**Comment by kabir** [comment]
@yersan the Preview Integration Failure is definitely unrelated

File: subsystem-test/tests/src/test/java/org/jboss/as/subsystem/test/transformationutils/TransformationUtilsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+//    public void testChildRegEmptySubsystemIsDefined() throws Exception {
+//        List<ModelNode> operations = new ArrayList<>();
+//        operations.add(Util.createAddOperation(SUBSYSTEM_ADDR));
+//        TransformationUtilsExtension.Builder builder = TransformationUtilsExtension.createBuilder();
+//        builder.
+//
+//        ModelNode model = runAndReadSubsystemModel(, operations);
+//
+//        Assert.assertTrue(model.isDefined());
+//    }
```
Comment (yersan) [inline subsystem-test/tests/src/test/java/org/jboss/as/subsystem/test/transformationutils/TransformationUtilsTestCase.java:378]: @kabir This is an unfinished test, can be removed or was it left pending to get it finished?

Comment (kabir) [reply subsystem-test/tests/src/test/java/org/jboss/as/subsystem/test/transformationutils/TransformationUtilsTestCase.java]: @yersan I got rid of it

**Comment by kabir** [comment]
Failures should be unrelated

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @kabir we plan to release it tomorrow

**Pull request merged**